src/index.js

```diff
@@ -0,0 +1,284 @@
+        return this[_connectToStatusNetwork](existingNetworkId.toString(), cb);
+      }
+      // otherwise, add a new network and connect to that network
+      return this[_addAndConnectStatusNetwork](cb);
+    });
+  }
+
+  /**
+   * Connects the Status app to a network specified by the `statusNetworkId`
+   * parameter.
+   * @private
+   * 
+   * @param {String} statusNetworkId ID that identifies the network in the Status 
+   * app to connect to.
+   * @param {Function} cb Callback called after the connect to network command 
+   * is sent to the Status app, called with parameters:
+   *  - {String} err Error that occurred connecting the Status app to the
+   *             network.
+   *  - {Object} result Result returned from the Status app on successful 
+   *             connection to the Embark network.
+   * 
+   * @returns {void}
+   */
+  [_connectToStatusNetwork](statusNetworkId, cb) {
+    this.logger.info(`Connecting Status app to network ${statusNetworkId}...`);
+    this.statusApi.connect(statusNetworkId, (err, result) => {
+      if (err) {
+        if (err.code === 'ECONNREFUSED' || err.code === 'ETIMEDOUT') {
+          return cb(`Failed to connect to the Status network. Is the Status app open?`);
+        }
+        return cb(`Error while connecting to Status network ${statusNetworkId} in the Status app: ${err.message}.`);
+      }
+      this.logger.info(`Successfully connected to network ${statusNetworkId}.`);
+      cb(null, result);
+    });
+  }
+
+  /**
+   * Adds a network to the Status app, then attempts to connect to it. Upon a
+   * successful addition of the network, the Status app will return a network ID,
+   * which is stored in the dApp's temporary filesystem to prevent future additions
+   * of an already exist network in the Status app.
+   * @private
+   * 
+   * @param {*} cb Callback called after the connect to network command 
+   * is sent to the Status app, called with parameters:
+   *  - {String} err Error that occurred adding and connecting a network.
+   *  - {Object} result Result returned from the Status app on successful 
+   *             connection to the Embark network.
+   * 
+   * @returns {void}
+   */
+  [_addAndConnectStatusNetwork](cb) {
+    // defined here because the configs have not been hydrated from the dapp when the plugin is instantiated
+    const nodePort = this.blockchainConfig.proxy ? this.blockchainConfig.rpcPort + 10 : this.blockchainConfig.rpcPort;
+    const nodeUrl = buildUrl('http', this.blockchainConfig.rpcHost, nodePort);
+    const networkName = `${NETWORK_NAME} (${this.pluginConfig.name})`;
+
+    this.logger.info(`Adding network '${networkName}' to Status...`);
+    return this.statusApi.addNetwork(networkName, nodeUrl, CHAIN_NAME, this.networkId, (err, result) => {
+      if (err) {
+        if (err.code === 'ECONNREFUSED' || err.code === 'ETIMEDOUT') {
+          return cb('Failed to add a network to the Status app. Is the Status app open?');
```

Comment:
I see this error message is used at least at another place. Do you think we could do some kind of method to handle errors? Or maybe just use a constant?

Reply:
Have pulled the error code out to a constant and the check to a static method.